src/statement.ts

```diff
@@ -101,12 +101,32 @@ export class Statement<S = unknown> {
 
   async #parseAndExtract(): Promise<ExtractedStatement> {
     const statementWithBindings = this.toString();
-    const { type, statements, tables } = await normalize(statementWithBindings);
+    const nameMap =
+      typeof this.config.aliases?.read === "function"
+        ? await this.config.aliases.read()
+        : undefined;
+    const { type, statements, tables } = await normalize(
+      statementWithBindings,
+      nameMap
+    );
+
+    // TODO: this feels wrong, but we have to use the normalize function to determine the type,
+    //    then use it again if the type is create to ensure that a new table isn't created with
+    //    an existing prefix.
+    if (type === "create" && nameMap != null) {
```

Comment:
`nameMap` can be `undefined`, but you are comparing to `null` here, so this will never match.

Reply:
When the typescript transpilation happens `!=` ends up ensuring the value is not `undefined` or `null`, which is needed for JS users I think.

Reply:
Ah ok, tricky. I guess I've never thought that deeply about transpiler behaviors and JS usage when writing typescript. Would be curious to know if, in general, this type of stuff is documented somewhere?

Reply:
Just to clarify, would it also be fine if you used `nameMap != undefined`? Would the transpiler still output a check for `undefined` or `null`? If so, I still think `nameMap != undefined` would be good here since it reads better in typescript and looks correct relative to our other typescript code that is in scope.

Reply:
I'm trying to rethink this with fresh eyes, and would it we better to check that `nameMap` is an Object?

RE: null vs. undefined
`== undefined` and `== null` are functionally identical.  They both check for `undefined` or `null`. I thought Typescript was transpiling `val == null` to `(val === null || val === undefined)`. But after looking in detail at the dist directory it looks like I was wrong.
Which to use feels like a style choice, and it would be nice to stay consistent.  In this package we use `== null` or `!= null` everywhere. I'm happy changing to using `undefined`, but maybe do it everywhere?  As a third option, we could also stop using [coercive comparisons](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality#comparison_with_type_conversion) all together?

Reply:
Yea seems like checking that it's an object makes sense.

Oh yea, coercive types... so that does actually work as expected, ignore my original comment. In general, I've always steered away from using coercive types because they cause this type of confusion and are black magic. Plus it's nice to have `null` and `undefined` mean different things in some cases. For example, `undefined` can mean "this has never been initialized" and `null` can mean "this was set to the empty value".